src/otx/engine/engine.py

```diff
@@ -327,6 +328,61 @@ def export(self, *args, **kwargs) -> None:
         """Export the trained model to OpenVINO Intermediate Representation (IR) or ONNX formats."""
         raise NotImplementedError
 
+    def explain(
+        self,
+        checkpoint: str | Path | None = None,
+        datamodule: EVAL_DATALOADERS | OTXDataModule | None = None,
+        explain_config: ExplainConfig | None = None,
+        **kwargs,
+    ) -> list | None:
+        """Run XAI using the specified model and data.
+
+        Args:
+            checkpoint (str | Path | None, optional): The path to the checkpoint file to load the model from.
+            datamodule (EVAL_DATALOADERS | OTXDataModule | None, optional): The data module to use for predictions.
+            explain_config (ExplainConfig | None, optional): Config used to handle saliency maps.
+            **kwargs: Additional keyword arguments for pl.Trainer configuration.
+
+        Returns:
+            list: Saliency maps.
+
+        Example:
+            >>> engine.explain(
+            ...     datamodule=OTXDataModule(),
+            ...     checkpoint=<checkpoint/path>,
+            ...     explain_config=ExplainConfig(),
+            ... )
+        """
+        from pathlib import Path
+
+        import cv2
+
+        ckpt_path = str(checkpoint) if checkpoint is not None else self.checkpoint
+        if explain_config is None:
+            explain_config = ExplainConfig()
+
+        lit_module = self._build_lightning_module(
+            model=self.model,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+        )
+        if datamodule is None:
+            datamodule = self.datamodule
+        lit_module.meta_info = datamodule.meta_info
+
+        lit_module.model.register_explain_hook()
+
+        self._build_trainer(**kwargs)
+
+        self.trainer.predict(
+            model=lit_module,
+            datamodule=datamodule,
+            ckpt_path=ckpt_path,
+        )
+        saliency_maps = self.trainer.model.model.explain_hook.records
+        cv2.imwrite(str(Path(self.work_dir) / "saliency_map.tiff"), saliency_maps[0][0])
```

Comment:
If it is an example of saving map, can you please add a comment like `Saving saliency map for the image 0, class 0`?

Reply:
Added